### PR TITLE
feat: add mega dropdown menu to nav

### LINKED ISSUE #100: Add "Documentation" dropdown to Navbar
### Feature description

We need to update the main navigation bar (aka _Navbar_) to replace the current "Documentation" link with a "Documentation" dropdown that shows various links to documentation pages.

We could try and use the ["Masthead" Carbon for IBM.com](https://www.ibm.com/standards/carbon/components/masthead) component if possible, or maybe at least take some inspiration from its implementation.

### Dependencies

This issue depends on:

- https://github.com/Qiskit/saiba/issues/134

### Ressources

- Design: https://www.figma.com/file/xc1jekL6jga6NyRH64k1mV/documentation-redesign?node-id=136%3A1821
- Prototype: https://www.figma.com/proto/xc1jekL6jga6NyRH64k1mV/documentation-redesign?page-id=152%3A1426&node-id=179%3A1819&viewport=335%2C-122%2C0.34&scaling=min-zoom&starting-point-node-id=179%3A1819&show-proto-sidebar=1

### Design capture

Screenshot 2022-08-24 at 10 51 30

### Additional context

- Please reach out to @JRussellHuffman for any questions on the design.
- _Carbon for IBM.com_'s _Masthead_ web component: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/tree/main/packages/web-components/src/components/masthead

### LINKED ISSUE #136: [Navbar] The Qiskit logo does not appear in Safari
### Steps to reproduce the problem

Just open the Navbar story:

https://qiskit.github.io/web-components/?path=/story/navbar--default

### What is the current behavior?

The Qiskit logo does not appear in Safari:

https://qiskit.github.io/web-components/?path=/story/navbar--default

### What is the expected behavior?

The Qiskit logo appear in Safari.

### Additional information

Extracted from: https://github.com/Qiskit/saiba/pull/53

### PR DESCRIPTION
**Pull request by yoshiemuranaka**
## Changes

Adds mega menu dropdown for the Documentation navigation item.

Resolves #100
Resolves #136 

## Implementation details
- Extended existing carbon header menu components with style overrides. Closely followed overrides in the ibmdotcom [masthead](https://www.ibm.com/standards/carbon/web-components/?path=/story/components-masthead--default) 
- Pulled as many documentation links I could find from live site (could use extra 👀 on the URLs)
- Includes mobile sidenav [design](https://www.figma.com/file/xc1jekL6jga6NyRH64k1mV/documentation-redesign?node-id=473%3A3171)

## How to read this PR
- Created wrappers for the mega menu that extend the carbon header menu components with additional styles

## Screenshots

**Final consolidated navigation changes**
Screen Shot 2022-09-08 at 9 20 48 AM

https://user-images.githubusercontent.com/8097306/188996122-5d726f86-88c7-4a36-a02f-35bf4247c701.mov

https://user-images.githubusercontent.com/8097306/188995955-31e43bbb-2b9a-4070-a649-90dc3957b553.mov

### CODE REVIEW

**Review by JRussellHuffman — CHANGES_REQUESTED** [review]
I'm really glad to see this moving forward, and it looks good. I have two minor pieces of feedback.
1. It looks like the color being used is Gray-10. Can it be swapped for Cool-gray-10?
2. The drop shadow under the component is too heavy. All we need is something very subtle.

Is it possible to see the component at smaller breakpoints? It looks good on the large size, but I can see it getting crowded on smaller sizes.

**Comment by yoshiemuranaka** [comment]
Screen Shot 2022-09-02 at 9 44 04 AM

^this would be the smallest screen size before the mobile layout

**Comment by javabster** [comment]
posting this here a 3rd time, can someone please respond 😅https://github.com/Qiskit/qiskit.org/issues/2702#issuecomment-1225894054

**Comment by yoshiemuranaka** [comment]
> sharing this a 3rd time, can someone please respond 😅[Qiskit/qiskit.org#2702 (comment)](https://github.com/Qiskit/qiskit.org/issues/2702#issuecomment-1225894054)

Hi @javabster sorry for the delayed response. @JRussellHuffman @gracelindsell could I get help with adding the additional nav menu items to the mega dropdown? Wondering if some of the submenu items could be consolidated from the original design, or if we want to add an additional 'row' underneath the existing menu options I can look into it..but may introduce scrolling inside the mega dropdown. Let me know what you think

**Comment by JRussellHuffman** [comment]
Here is a screenshot with the new column for contributing added.

Screen Shot 2022-09-07 at 1 51 12 PM

It looks like the content items have been updated a little bit, and I agree with Yoshie, we may want to consider some consolidation. Here is one option
* Consolidate partners and providers. The only option under partners was "all providers". If that wasn't a spelling error, then consolidating those two seems easy.
* We could also consider grouping the textbook with the other help resources and changing the label to "Help and Learning"
* Lastly, we could possibly combine Overview and Qiskit Terra, but I'm not sure if this is a good idea.

Adding @1ucian0 as FYI.

**Comment by yoshiemuranaka** [comment]
Thanks @JRussellHuffman, I think the consolidated version looks great. Here is a screenshot of the changes at the smallest breakpoint

Screen Shot 2022-09-07 at 12 10 29 PM

I'll wait for consensus before making changes in this PR

**Comment by JRussellHuffman** [comment]
Looks good. What determines the column width? I only ask because the Qiskit Terra link is probably the most important link and it also happens to be one of the few that wraps. Would it be possible to give that one more space and instead something like "Applications" or "Help and Learning" gets a little less?

File: components/ui-shell/settings.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      {
+        label: 'Get Help',
+        children: [
+          {
+            label: 'Stack Exchange',
+            url: 'https://qiskit.org/documentation/', //TODO: update
+          },
+          {
+            label: 'qiskit-community-feedback',
+            url: 'https://qiskit.org/documentation/', //TODO: update
```
Comment (techtolentino) [inline components/ui-shell/settings.ts:153]: Should qiskit-community-feedback point to https://github.com/Qiskit/feedback ?

File: components/ui-shell/settings.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            url: 'https://qiskit.org/learn/',
+          },
+        ],
+      },
+      {
+        label: 'Get Help',
+        children: [
+          {
+            label: 'Stack Exchange',
+            url: 'https://qiskit.org/documentation/', //TODO: update
```
Comment (techtolentino) [inline components/ui-shell/settings.ts:149]: Should Stack Exchange point to https://quantumcomputing.stackexchange.com/ ? @javabster

**Comment by yoshiemuranaka** [comment]
> Looks good. What determines the column width? I only ask because the Qiskit Terra link is probably the most important link and it also happens to be one of the few that wraps. Would it be possible to give that one more space and instead something like "Applications" or "Help and Learning" gets a little less?

I updated the code to use a [grid](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout) layout which will ensure that each menu section/column is a consistent width. Before that I was using a [flex](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Basic_Concepts_of_Flexbox) display which tried to evenly space the items within the available space while trying to respect natural content width. We can also adjust each column width as needed.

**Before with flex**
Screen Shot 2022-09-07 at 3 50 26 PM


**After with grid**
Screen Shot 2022-09-07 at 3 49 42 PM

File: components/ui-shell/settings.ts (hunk 1 above)
Comment (HuangJunye) [reply components/ui-shell/settings.ts:153]: Yes I think that's the correct link. We recently moved feedback repo from qiskit-community github org to qiskit github org. Should we change the label as well? Should we use `Qiskit Feedback`?

File: components/ui-shell/settings.ts (hunk 2 above)
Comment (HuangJunye) [reply components/ui-shell/settings.ts:149]: I think we should use this link: https://quantumcomputing.stackexchange.com/questions/tagged/qiskit, same as in  the footer: https://github.com/Qiskit/qiskit.org/blob/3e1c1abfb8f50977146eb5da0343c600064aa897/constants/menuLinks.ts#L59-L60

File: components/ui-shell/settings.ts (hunk 1 above)
Comment (techtolentino) [reply components/ui-shell/settings.ts:153]: Thanks @HuangJunye

File: components/ui-shell/settings.ts (hunk 2 above)
Comment (javabster) [reply components/ui-shell/settings.ts]: agree with junye!

**Comment by javabster** [comment]
this is looking great, thank-you for the speedy changes @yoshiemuranaka

**Comment by yoshiemuranaka** [comment]
Thanks @javabster @JRussellHuffman! I added a screenshot of navigation changes in the PR description

File: components/ui-shell/settings.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      {
+        label: 'Advocates',
+        url: 'https://qiskit.org/advocates/',
+      },
+    ],
+  },
+  {
+    label: 'Documentation',
+    isMegaMenu: true,
+    url: 'https://qiskit.org/documentation/',
```
Comment (HuangJunye) [inline components/ui-shell/settings.ts:47]: Does the “Documentation” tab still linked to the homepage or it only expands to show the top menu?

File: components/ui-shell/settings.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            url: 'https://qiskit.org/documentation/getting_started.html',
+          },
+        ],
+      },
+      {
+        label: 'Core',
+        children: [
+          {
+            label: 'Qiskit Terra',
+            url: 'https://qiskit.org/documentation/apidoc/terra.html',
```
Comment (HuangJunye) [inline components/ui-shell/settings.ts:63]: Ideally we should link to the module homepage with overview like ([qiskit-nature](https://qiskit.org/documentation/nature/)) but I know for Terra, Aer and IBMQ provider that are still under the meta package and therefore do not have their individual homepages this is not possible.

But Terra is effectively Qiskit. Perhaps this link should be linking to the Qiskit Documentation homepage: https://qiskit.org/documentation/

With Aer 0.11 release (coming soon), aer documentation will have [it's own page](https://github.com/Qiskit/qiskit/pull/1595). Depending on when this PR is merged, we should change the link to https://qiskit.org/documentation/aer

**Review by HuangJunye — COMMENTED** [review]
Looks great! Thanks for the good work. I left a few comments about the links.

**Review by techtolentino — COMMENTED** [review]
After seeing this locally and testing it for a11y tab-ability, I really like this component! 

My only question is what Leron pointed out at sprint review, regarding having the User Account button as part of the `Default` variant, and one w/o as the alternative. 

Otherwise, 👏 this is great!

**Comment by yoshiemuranaka** [comment]
> After seeing this locally and testing it for a11y tab-ability, I really like this component!
> 
> My only question is what Leron pointed out at sprint review, regarding having the User Account button as part of the `Default` variant, and one w/o as the alternative.
> 
> Otherwise, 👏 this is great!

Thanks @techtolentino! updated to make the nav with account icon the default

**Review by techtolentino — APPROVED** [review]
I'm good w/ this 👏

**Comment by abdonrd** [comment]
I'm late reading PRs, but... Great job, @yoshiemuranaka! 👏

**Comment by JRussellHuffman** [comment]
Looks good to me! Thanks for the quick and good work on this one.

**Pull request merged**